src/feature/hs/hs_intropoint.c

```diff
@@ -182,6 +182,60 @@ hs_intro_send_intro_established_cell,(or_circuit_t *circ))
   return ret;
 }
 
+/* Validate the cell DoS extension parameters. Return true iff they've been
+ * bound check and can be used. Else return false. See proposal 305 for
+ * details and reasons about this validation. */
+STATIC bool
+validate_cell_dos_extension_parameters(uint64_t intro2_rate_per_sec,
+                                       uint64_t intro2_burst_per_sec)
+{
+  bool ret = false;
+
+  /* Bound checking the received rate per second. As per proposal 305, a
+   * value of 0 disabled the defenses regardless of the other parameters in
+   * the extension which is why we stop right away. */
+  if (intro2_rate_per_sec > HS_CONFIG_V3_DOS_DEFENSE_RATE_PER_SEC_MAX ||
+      intro2_rate_per_sec <= HS_CONFIG_V3_DOS_DEFENSE_RATE_PER_SEC_MIN) {
+    if (intro2_rate_per_sec != 0) {
```

Comment:
What's the point of this explicit `intro2_rate_per_sec != 0` check? Can't we just do the log even if it's 0?

Reply:
0 is valid value. It means "disable defense" per prop305. So instead of logging "bad value", we just skip it.